infra/cifuzz/filestore/gitlab/__init__.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Empty filestore implementation for platforms that haven't implemented it."""
+import logging
+
+import shutil
+import os
+import json
+import tempfile
+
+import http_utils
+import filestore
+
+# pylint:disable=no-self-use,unused-argument
+
+
+class GitlabFilestore(filestore.BaseFilestore):
+  """Implementation of BaseFilestore using Gitlab job artifacts. Relies on
+  using a cache to get latest job id oto get a type of artifacts.
+  having a PRIVATE-TOKEN supplied to access the Gitlab API as seems
+  the only way to download an artifact from another job"""
+
+  def __init__(self, config):
+    super().__init__(config)
+    self.downloaded = set()
+    self.artifacts_dir = self.config.platform_conf.artifacts_dir
+    self.cache_dir = self.config.platform_conf.cache_dir
```

Comment:
Does cache_dir need to be in config at all? If it's only used in this file, maybe we can move it here.
Why does it need to be defined by the environment at all. Why can't it be a temp directory that gets destroyed in this object's destructor?

Reply:
`cache_dir` needs to match the one defined in .gitlab-ci.yml (as for `artifacts_dir`)
We could set it to a hardcoded value...
I felt it was more user-friendly to let it be configurable...

> Why can't it be a temp directory that gets destroyed in this object's destructor?

This cache directory must not be deleted.
After the job is successful, gitlab will the content of the directory to update the cache for real.

Reply:
>I felt it was more user-friendly to let it be configurable...

Can you explain how a user might find it helpful to set this?
I'm completely open to hearing why since I don't really understand the purpose of the cache, but right now this doesn't seem useful and seems to add some complexity IMO.